doc/nvim-tree-lua.txt

```diff
@@ -949,6 +957,43 @@ Configuration options for tree indent markers.
           none = " ",
         }
 <
+*nvim-tree.renderer.size*
+Configuration for file size display column
+
+    *nvim-tree.renderer.size.enable*
+    Determine if we should render file size information.
+      Type: `boolean`, Default: `false`
+
+    *nvim-tree.renderer.size.width_cutoff*
+    Whenever nvim-tree window has less width than this cutoff the
+    size information is hidden. If it goes back to having bigger width
+    then the cutoff, the information is shown again.
+      Type: `integer`, Default: `18`
+
+    *nvim-tree.renderer.size.column_width*
+    Determines the amount of space (in characters) the
+    file size information will take.
+      Type: `integer`, Default: `10`
+
+    *nvim-tree.renderer.size.show_folder_size*
```

Comment:
I'm not sure how much value this is; it's only showing the fs_stat size - the folder's size, rather than the content size.

Could we just leave folders out completely?

Reply:
As of right now there's no value indeed, but I'd like to support content size information.
What do you think two approaches to getting content size of a folder:
1) recursively calculate once the content size for each folder, and then update folder fize information as files are moved or changed
2) call external ``df -h`` (there must the something like it for windows) on the folder and render that.

Reply:
> 2\. call external `df -h -s` (there must the something like it for windows) on the folder and render that.

That would be really transparent and a good default for most users. It could be a user configurable like `trash.cmd` so that windows users could do whatever they need to do, and linux/macos users could use an alternate like `tree`

> 1. recursively calculate once the content size for each folder, and then update folder fize information as files are moved or changed

That would work well as it would sidestep any performance issues:
The watchers get a notification for the directory itself so they could do it in the async job. We'd just need to recalculate folders up the tree afterwards.
`R` could calculate everything. It's synchronous, but we know that a full refresh takes time.

Reply:
Do you want to do folders in a follow up PR? File sizes is pretty much ready to go...